consensus/setup.go

```diff
@@ -0,0 +1,74 @@
+package consensus
+
+import (
+	"sync"
+	"time"
+
+	"github.com/bitmark-inc/bitmarkd/background"
+	"github.com/bitmark-inc/bitmarkd/fault"
+	"github.com/bitmark-inc/bitmarkd/p2p"
+	"github.com/bitmark-inc/logger"
+)
+
+var globalData Consensus
+
+const (
+	votingMetricRunInitial  = 60 * time.Second // should reference announce and p2p initial
+	votingMetricRunInterval = 30 * time.Second
+	machineRunInitial       = 70 * time.Second // should reference announce and p2p initial
+	machineRunInterval      = 15 * time.Second
+)
+
+//Consensus is a wrap struct for consensus  state machine
+type Consensus struct {
+	sync.RWMutex            // to allow locking
+	Log           *logger.L // logger
+	Node          *p2p.Node
+	machine       Machine
+	votingMetrics MetricsPeersVoting
+	initialised   bool
+	background    *background.T
+}
+
+// Initialise consensus package
+func Initialise(node *p2p.Node, fastsync bool) error {
+	globalData.Lock()
+	defer globalData.Unlock()
+	if globalData.initialised {
+		return fault.AlreadyInitialised
+	}
+	if nil == node {
```

Comment:
In `command/bitmarkd/main.go` line 314, it has check to make sure `node` is not nil, does it need to check again here?

Reply:
A package should not know what's going on outside the package. So I think consensus needs to make sure the p2p node is not  nil.

Reply:
That's what I think, currently `bitmarkd/main` needs to check `p2p` status, if package is concealed, then bitmarkd should only check return error of `p2p.Initialise` instead of checking by other function (`p2p.P2PNode == nil`)

Reply:
Generally, I agree with your idea. However, it is a kind of coupling if we do wait in the consensus. I think it should also check but not wait.

Reply:
or we should initialize P2P inside concensus ...